volmdlr/faces.py

```diff
@@ -7557,6 +7561,138 @@ def plot(self, ax=None, color: str = 'k', alpha: float = 1):
 
         return ax
 
+    def project_coincident_faces_of(self, shell):
+        """
+        Divides self's faces based on coincident shell's faces
+        """
+
+        used_faces, list_faces = {}, []
+        initial_faces = self.faces[:]
+
+        for i, face1 in enumerate(initial_faces):
+            for j, face2 in enumerate(shell.faces):
+                contour1 = face1.surface2d.outer_contour
```

Comment:
Could we make this part of the code a method of face taking another face into parameter?

Reply:
@masfaraud 
I tried to define a new method for PlaneFace3D which takes a list of faces (it is more significant than just one face) -> here in this [**commit**](https://github.com/Dessia-tech/volmdlr/pull/463/commits/5f4350fe93bb1b8f23a9d7257f5082d128a7d8c2)